services/consensusalgo/leanhelixconsensus/block_provider.go

```diff
@@ -7,15 +7,112 @@ import (
 	"github.com/orbs-network/orbs-network-go/crypto/digest"
 	"github.com/orbs-network/orbs-network-go/crypto/logic"
 	"github.com/orbs-network/orbs-network-go/crypto/signature"
-	"github.com/orbs-network/orbs-network-go/instrumentation/log"
 	"github.com/orbs-network/orbs-spec/types/go/primitives"
 	"github.com/orbs-network/orbs-spec/types/go/protocol"
 	"github.com/orbs-network/orbs-spec/types/go/protocol/consensus"
 	"github.com/orbs-network/orbs-spec/types/go/services"
 	"github.com/orbs-network/orbs-spec/types/go/services/handlers"
 	"github.com/pkg/errors"
+	"time"
 )
 
+type BlockPairWrapper struct {
+	blockPair *protocol.BlockPairContainer
+}
+
+func (b *BlockPairWrapper) Height() lhprimitives.BlockHeight {
+	return lhprimitives.BlockHeight(b.blockPair.TransactionsBlock.Header.BlockHeight())
+}
+
+func (b *BlockPairWrapper) BlockHash() lhprimitives.Uint256 {
+	// TODO This is surely incorrect, fix to use the right hash
+	return lhprimitives.Uint256(b.blockPair.TransactionsBlock.Header.MetadataHash())
+}
+
+func NewBlockPairWrapper(blockPair *protocol.BlockPairContainer) *BlockPairWrapper {
+	return &BlockPairWrapper{
+		blockPair: blockPair,
+	}
+}
+
+// TODO This should be updated on commit!!!
+// Currently put the state of the last committed block here, but it might not be the right place for it.
+// See https://tree.taiga.io/project/orbs-network/us/404
+type blockProvider struct {
+	consensusContext              services.ConsensusContext
+	consensusRoundTimeoutInterval time.Duration
+	nodePublicKey                 primitives.Ed25519PublicKey
+	nodePrivateKey                primitives.Ed25519PrivateKey
+	lastCommittedBlock            *protocol.BlockPairContainer
+	lastCommittedBlockHeight      primitives.BlockHeight
+}
+
+func NewBlockProvider(consensusRoundTimeoutInterval time.Duration, nodePublicKey primitives.Ed25519PublicKey, nodePrivateKey primitives.Ed25519PrivateKey) *blockProvider {
```

Comment:
config can change in runtime so it's better to pass the function rather than the result of invoking it